pillar/vault.py

```diff
@@ -169,6 +169,33 @@ def _authenticate(conn):
         conn.token = os.environ.get('VAULT_TOKEN')
 
 
+def couple(location, conn):
+    """
+    If location is a dictionary, return a dictionary of k->v pairs looked up from vault.
+    If location is a string, return the value looked up from vault.
+    """
```

Comment:
maybe something about recursively walking dictionaries to handle namespacing in the secrets file?